src/components/ProtectedRoute.jsx

```diff
@@ -1,95 +0,0 @@
-import React, { useContext, useEffect, useState } from 'react'
-import { Redirect, Route } from 'react-router-dom'
-import axios from 'axios'
-
-// Import layouts
-import Home from '../layouts/Home';
-
-// Import contexts
-const { LoginContext } = require('../contexts/LoginContext')
-
-// Constants
-const authContextString = '#isomercms'
-const userIdKey = 'userId'
-
-// axios settings
-axios.defaults.withCredentials = true
-
-const ProtectedRoute = ({ component: WrappedComponent, ...rest }) => {
-    const { isLoggedIn } = useContext(LoginContext)
-    const [pageNotFound, setPageNotFound] = useState()
-    const { siteName } = rest.computedMatch?.params
-    
-    useEffect(() => {
-        let _isMounted = true
-        const fetchData = async () => {
-            try {
-                await axios.get(`${process.env.REACT_APP_BACKEND_URL}/sites/${siteName}`)
-                if (_isMounted) setPageNotFound(false)
-            } catch (e) {
-                if (_isMounted) setPageNotFound(true)
-            }
-        }
-        if (siteName) {
-            fetchData()
-        } else {
-            if (_isMounted) setPageNotFound(false)
-        }
```

Comment:
I missed this specific feature out - though I think it should be handled within the `Workspace` layout instead of the `RouteSelector` component. This is currently done for several other layouts like `EditPage` and `EditNavBar`.

We should ideally keep the `RouteSelector` component's job strictly to routing to the specific layouts/components based on the route pathname. The specific layouts/components should handle the retrieval of data through an API, and if that API fails, the components call a redirection to the `not-found` route pathname.

Created an issue https://github.com/isomerpages/isomercms-frontend/issues/434

Reply:
I agree that we should restrict the route selector's job to simply routing. The reason why we had this check in `ProtectedRoute` is because this check applies for all repo-specific layouts (basically all the layouts except for `Sites` and `Home`) instead of implementing them one by one. Perhaps we can create another higher-order component which renders all site-specific layouts as children while also running this useEffect check? So

```
<RouteSelector>
  <Home />
  <Sites />
  <HOC>
    < ... all site-specific layouts>
  </ HOC>
</RouteSelector>
```

Apologize for formatting errors, I was on mobile